File: frontend/src/api/ai.ts
import api from "./client";

export const createSession = (title?: string) =>
  api.post("/ai/sessions", { title }).then((r) => r.data);

export const listSessions = () =>
  api.get("/ai/sessions").then((r) => r.data);

export const chat = (sessionId: number, message: string) =>
  api.post("/ai/chat", { session_id: sessionId, message }).then((r) => r.data);

export const investigate = (evidence: string) =>
  api.post("/ai/investigate", { evidence }).then((r) => r.data);

export const getRecommendations = (logId: number) =>
  api.post("/ai/recommendations", { log_id: logId }).then((r) => r.data);

export const explainLog = (text: string) =>
  api.post("/ai/explain", { text }).then((r) => r.data);

// ── AI Providers ──

export interface AIProviderInfo {
  name: string;
  available: boolean;
  requires_api_key: boolean;
}

export const listProviders = () =>
  api.get<AIProviderInfo[]>("/ai/providers").then((r) => r.data);

export const getActiveProvider = () =>
  api.get<AIProviderInfo>("/ai/providers/active").then((r) => r.data);

export const switchProvider = (provider: string) =>
  api.post<AIProviderInfo>("/ai/providers/switch", { provider }).then((r) => r.data);

File: src/schemas/ai.py
from pydantic import BaseModel


class ChatSessionCreate(BaseModel):
    title: str = "New Chat"
    context_type: str = "general"
    context_id: int | None = None


class ChatSessionResponse(BaseModel):
    id: int
    title: str
    context_type: str
    context_id: int | None = None
    created_at: str
    updated_at: str


class ChatMessage(BaseModel):
    role: str
    content: str
    created_at: str | None = None


class ChatRequest(BaseModel):
    session_id: int
    message: str
    stream: bool = False


class ChatResponse(BaseModel):
    session_id: int
    reply: str
    evidence: list[dict] = []
    suggested_questions: list[str] = []
    confidence: float = 1.0


class ExplainRequest(BaseModel):
    text: str
    context_type: str = "general"


class ExplainResponse(BaseModel):
    explanation: str
    confidence: float
    sources: list[str] = []


class InvestigationRequest(BaseModel):
    log_id: int | None = None
    evidence: str = ""


class InvestigationResponse(BaseModel):
    summary: str
    suspicious_patterns: list[dict] = []
    critical_events: list[dict] = []
    attack_chain: list[str] = []
    priorities: list[dict] = []
    recommendations: list[str] = []
    confidence: float = 1.0


class RecommendationResponse(BaseModel):
    next_steps: list[str] = []
    containment: list[str] = []
    recovery: list[str] = []
    patching: list[str] = []
    detection_improvements: list[str] = []


class AIProviderInfo(BaseModel):
    name: str
    available: bool
    requires_api_key: bool


class TimelineAnalysisResponse(BaseModel):
    timeline: list[dict] = []
    attack_stages: list[str] = []
    root_cause: str = ""
    confidence: float = 1.0


class PromptTemplate(BaseModel):
    id: int | None = None
    name: str
    category: str
    content: str
    version: str = "1.0"
    is_active: bool = True
